diff --git a/srsue/hdr/stack/mac_nr/proc_sr_nr.h b/srsue/hdr/stack/mac_nr/proc_sr_nr.h
index caa2a635c..319779e4e 100644
--- a/srsue/hdr/stack/mac_nr/proc_sr_nr.h
+++ b/srsue/hdr/stack/mac_nr/proc_sr_nr.h
@@ -50,8 +50,9 @@ public:
 
 private:
   void reset_nolock();
-  int  sr_counter    = 0;
-  bool is_pending_sr = false;
+  int  sr_counter          = 0;
+  int  sr_prohibit_counter = 0;
+  bool is_pending_sr       = false;
 
   srsran::sr_cfg_nr_t cfg = {};
 
