diff --git a/examples/xApp/c/monitor/CMakeLists.txt b/examples/xApp/c/monitor/CMakeLists.txt
index c109188e..1e75b6cd 100644
--- a/examples/xApp/c/monitor/CMakeLists.txt
+++ b/examples/xApp/c/monitor/CMakeLists.txt
@@ -36,3 +36,16 @@ target_link_libraries(xapp_rc_moni
                      -lsctp
                      -ldl
                      )
+
+add_executable(xapp_kpm_moni_write_to_csv 
+		xapp_kpm_moni_write_to_csv.c
+                ../../../../src/util/alg_ds/alg/defer.c
+              )
+
+target_link_libraries(xapp_kpm_moni_write_to_csv
+                    PUBLIC
+                    e42_xapp
+                    -pthread
+                    -lsctp
+                    -ldl
+                    )
\ No newline at end of file
