diff --git a/examples/xApp/c/monitor/xapp_kpm_moni.c b/examples/xApp/c/monitor/xapp_kpm_moni.c
index 2918080e..52b8c5fc 100644
--- a/examples/xApp/c/monitor/xapp_kpm_moni.c
+++ b/examples/xApp/c/monitor/xapp_kpm_moni.c
@@ -276,16 +276,63 @@ void log_kpm_measurements(kpm_ind_msg_format_1_t const* msg_frm_1)
   for (size_t j = 0; j < msg_frm_1->meas_data_lst_len; j++) {
     meas_data_lst_t const data_item = msg_frm_1->meas_data_lst[j];
 
+    size_t rec_idx = 0;
     for (size_t i = 0; i < msg_frm_1->meas_info_lst_len; i++) {
       const meas_info_format_1_lst_t info_item = msg_frm_1->meas_info_lst[i];
-      for (size_t z = 0; z < info_item.label_info_lst_len; z++) {
-        const label_info_lst_t label_info = info_item.label_info_lst[z];
-        const meas_record_lst_t record_item = data_item.meas_record_lst[i + z];
 
-        match_meas_type[info_item.meas_type.type](info_item.meas_type, label_info, record_item);
+      if (info_item.label_info_lst_len > 1 && info_item.meas_type.type == NAME_MEAS_TYPE && info_item.label_info_lst[0].distBinX != NULL) {
+        char *name_str = cp_ba_to_str(info_item.meas_type.name);
+        char *name_unit = get_meas_unit(name_str);
+        if (name_unit && strcmp(name_unit, "[]") == 0) name_unit = "";
+        if (name_unit == NULL) name_unit = "";
+
+        printf("%s = [", name_str);
+        uint32_t last_x = 0, last_y = 0;
+
+        for (size_t z = 0; z < info_item.label_info_lst_len; z++) {
+
+          const label_info_lst_t label_info = info_item.label_info_lst[z];
+          const meas_record_lst_t record_item = data_item.meas_record_lst[rec_idx++];
+
+          uint32_t cur_x = *label_info.distBinX;
+          uint32_t cur_y = *label_info.distBinY;
+
+          if (z == 0) {
+            printf("[[");
+          } else {
+            if (cur_x != last_x) {
+              printf("]], [[");
+            } else if (cur_y != last_y) {
+              printf("], [");
+            } else {
+              printf(", ");
+            }
+          }
+          if (record_item.value == 0) {
+            printf("%d", record_item.int_val);
+          } else if (record_item.value == 1) {
+            printf("%.2f", record_item.real_val);
+          } else {
+            printf("null");
+          }
+          last_x = cur_x;
+          last_y = cur_y;
+          if (data_item.incomplete_flag && *data_item.incomplete_flag == TRUE_ENUM_VALUE) {
+            printf(" /* Record not reliable */");
+          }
+        }
+        printf("]]] %s\n", name_unit);
+        free(name_str);
+      } else {
+        for (size_t z = 0; z < info_item.label_info_lst_len; z++) {
+          const label_info_lst_t label_info = info_item.label_info_lst[z];
+          const meas_record_lst_t record_item = data_item.meas_record_lst[rec_idx++];
 
-        if (data_item.incomplete_flag && *data_item.incomplete_flag == TRUE_ENUM_VALUE)
-          printf("Measurement Record not reliable");
+          match_meas_type[info_item.meas_type.type](info_item.meas_type, label_info, record_item);
+
+          if (data_item.incomplete_flag && *data_item.incomplete_flag == TRUE_ENUM_VALUE)
+            printf("Measurement Record not reliable\n");
+        }
       }
     }
   }
@@ -404,6 +451,9 @@ label_info_lst_t fill_kpm_label(void)
   return label_item;
 }
 
+static
+label_info_lst_t fill_distribution_bin_label(const uint32_t x, const uint32_t y, const uint32_t z);
+
 static
 kpm_act_def_format_1_t fill_act_def_frm_1(ric_report_style_item_t const* report_item)
 {
@@ -427,9 +477,23 @@ kpm_act_def_format_1_t fill_act_def_frm_1(ric_report_style_item_t const* report_
 
     // [1, 2147483647]
     // 8.3.11
-    meas_item->label_info_lst_len = 1;
-    meas_item->label_info_lst = ecalloc(1, sizeof(label_info_lst_t));
-    meas_item->label_info_lst[0] = fill_kpm_label();
+    if (cmp_str_ba("CARR.PDSCHMCSDist", meas_item->meas_type.name) == 0) {
+      /// 1-8 RI, 1-3 MCS table, 0-31 MCS value
+      meas_item->label_info_lst_len = 8 * 3 * 32;
+      meas_item->label_info_lst = ecalloc(meas_item->label_info_lst_len, sizeof(label_info_lst_t));
+      size_t idx = 0;
+      for (uint32_t x = 1; x <= 8; x++) {
+        for (uint32_t y = 1; y <= 3; y++) {
+          for(uint32_t z = 0; z <= 31; z++) {
+            meas_item->label_info_lst[idx++] = fill_distribution_bin_label(x, y, z);
+          }
+        }
+      }
+    } else {
+      meas_item->label_info_lst_len = 1;
+      meas_item->label_info_lst = ecalloc(meas_item->label_info_lst_len, sizeof(label_info_lst_t));
+      meas_item->label_info_lst[0] = fill_kpm_label();
+    }
   }
 
   // 8.3.8 [0, 4294967295]
