diff --git a/lib/test/asn1/rrc_nr_utils_test.cc b/lib/test/asn1/rrc_nr_utils_test.cc
index 9f551bce1..c6820104b 100644
--- a/lib/test/asn1/rrc_nr_utils_test.cc
+++ b/lib/test/asn1/rrc_nr_utils_test.cc
@@ -75,6 +75,22 @@ int test_mac_rach_common_config()
   TESTASSERT(rach_cfg_nr.PreambleReceivedTargetPower == -110);
   TESTASSERT(rach_cfg_nr.preambleTransMax == 7);
   TESTASSERT(rach_cfg_nr.powerRampingStep == 4);
+  TESTASSERT(rach_cfg_nr.nof_preambles == 64);
+
+  rach_common_config_asn1.ssb_per_rach_occasion_and_cb_preambs_per_ssb_present = true;
+  TESTASSERT(asn1::number_to_enum(rach_common_config_asn1.ssb_per_rach_occasion_and_cb_preambs_per_ssb.set_one(),
+                                  64));
+
+  rach_cfg_nr = {};
+  make_mac_rach_cfg(rach_common_config_asn1, &rach_cfg_nr);
+  TESTASSERT(rach_cfg_nr.nof_preambles == 64);
+
+  rach_common_config_asn1.total_nof_ra_preambs_present = true;
+  rach_common_config_asn1.total_nof_ra_preambs         = 8;
+
+  rach_cfg_nr = {};
+  make_mac_rach_cfg(rach_common_config_asn1, &rach_cfg_nr);
+  TESTASSERT(rach_cfg_nr.nof_preambles == 64);
   return SRSRAN_SUCCESS;
 }
 
