diff --git a/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm.c b/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm.c
index c9620e871f..5cefdc122a 100644
--- a/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm.c
+++ b/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm.c
@@ -426,6 +426,23 @@ static const char* kpm_meas_du[] = {
   "DRB.UEThpUl",
   "RRU.PrbTotDl",
   "RRU.PrbTotUl",
+  "RSRP",
+  "N_RSRP_MEAS",
+  "N_PRB",
+  "RSSI",
+  "RSRQ",
+  "PUSCH_SNR",
+  "PUCCH_SNR",
+  "DRB.HarqMcsUl",
+  "DRB.HarqMcsDl",
+  "DRB.HarqBlockErrorRateUl",
+  "DRB.HarqBlockErrorRateDl",
+  "DRB.MacSduRetransmissionRateUl",
+  "DRB.MacSduRetransmissionRateDl",
+  "DRB.MacSduErrorRateUl",
+  "DRB.MacSduErrorRateDl",
+  "CQI_SINGLE_CODEWORD",
+  "CQI_DUAL_CODEWORD",
   NULL,
 };
 
@@ -437,6 +454,23 @@ static const char* kpm_meas_gnb[] = {
   "DRB.UEThpUl",
   "RRU.PrbTotDl",
   "RRU.PrbTotUl",
+  "RSRP",
+  "N_RSRP_MEAS",
+  "N_PRB",
+  "RSSI",
+  "RSRQ",
+  "PUSCH_SNR",
+  "PUCCH_SNR",
+  "DRB.HarqMcsUl",
+  "DRB.HarqMcsDl",
+  "DRB.HarqBlockErrorRateUl",
+  "DRB.HarqBlockErrorRateDl",
+  "DRB.MacSduRetransmissionRateUl",
+  "DRB.MacSduRetransmissionRateDl",
+  "DRB.MacSduErrorRateUl",
+  "DRB.MacSduErrorRateDl",
+  "CQI_SINGLE_CODEWORD",
+  "CQI_DUAL_CODEWORD",
   NULL,
 };
 
