diff --git a/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm_subs.h b/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm_subs.h
index e612ca2bab..f2ef736f5f 100644
--- a/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm_subs.h
+++ b/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm_subs.h
@@ -48,6 +48,9 @@ typedef struct{
 } kv_measure_t;
 
 
+void set_kpm_label_info(const label_info_lst_t* label_info);
+const label_info_lst_t* get_kpm_label_info(void);
+
 void init_kpm_subs_data(void);
 
 meas_record_lst_t get_kpm_meas_value(char* kpm_meas_name, uint32_t gran_period_ms, cudu_ue_info_pair_t ue_info, const size_t ue_idx, e2_node_level_stats_t* node_stats);
