diff --git a/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm.c b/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm.c
index c9620e871f..52b8efe9e1 100644
--- a/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm.c
+++ b/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm.c
@@ -437,6 +437,15 @@ static const char* kpm_meas_gnb[] = {
   "DRB.UEThpUl",
   "RRU.PrbTotDl",
   "RRU.PrbTotUl",
+  "RSRP",
+  "RSRQ",
+  "N_RSRP_MEAS",
+  "N_PRB",
+  "PUSCH_SNR",
+  "PUCCH_SNR",
+  "RAW_RSSI",
+  "UL_RSSI",
+  "CQI",
   NULL,
 };
 
