diff --git a/lib/mac/mac_dl/mac_cell_processor.cpp b/lib/mac/mac_dl/mac_cell_processor.cpp
index c5ecf0492b..432618aa5a 100644
--- a/lib/mac/mac_dl/mac_cell_processor.cpp
+++ b/lib/mac/mac_dl/mac_cell_processor.cpp
@@ -11,6 +11,7 @@
 #include "ocudu/ran/band_helper.h"
 #include "ocudu/ran/pdsch/pdsch_constants.h"
 #include "ocudu/scheduler/result/sched_result.h"
+#include "ocudu/support/async/async_no_op_task.h"
 #include "ocudu/support/async/async_timer.h"
 #include "ocudu/support/async/execute_on_blocking.h"
 #include "ocudu/support/executors/execute_until_success.h"
@@ -72,23 +73,14 @@ async_task<void> mac_cell_processor::start()
   // Note: This is done in the control executor context to avoid concurrency with other CTRL procedures.
   sched.handle_cell_activation(cell_cfg.cell_index);
 
-  return execute_and_continue_on_blocking(
-      cell_exec,
-      ctrl_exec,
-      timers,
-      [this]() noexcept OCUDU_RTSAN_NONBLOCKING {
-        if (state != cell_state::inactive) {
-          // No-op.
-          return;
-        }
+  if (state != cell_state::inactive) {
+    return launch_no_op_task();
+  }
 
-        state = cell_state::active;
-        logger.info("cell={}: Cell was activated", fmt::underlying(cell_cfg.cell_index));
-      },
-      [this]() {
-        logger.warning("cell={}: Postponed cell start operation. Cause: Task queue is full",
-                       fmt::underlying(cell_cfg.cell_index));
-      });
+  state = cell_state::active;
+  logger.info("cell={}: Cell was activated", fmt::underlying(cell_cfg.cell_index));
+
+  return launch_no_op_task();
 }
 
 async_task<void> mac_cell_processor::stop()
@@ -215,32 +207,21 @@ async_task<bool> mac_cell_processor::add_ue(const mac_ue_create_request& request
   // > Create a MAC UE DL context.
   mac_dl_ue_context ue_inst(request);
 
-  // > Update the UE context inside the cell executor.
-  return defer_and_continue_on_blocking(
-      cell_exec,
-      ctrl_exec,
-      timers,
-      [this, ue_inst = std::move(ue_inst)]() mutable noexcept OCUDU_RTSAN_NONBLOCKING {
-        // > Insert UE and DL bearers.
-
-        // Note: Ensure we only do so if the cell is active.
-        if (state != cell_state::active) {
-          // Note: We do not need to call dl_harq_buffers.deallocate_ue_buffers(...) because the buffers were reset
-          // on cell deactivation.
-          return false;
-        }
+  // > Insert UE and DL bearers.
+  // Note: Ensure we only do so if the cell is active.
+  if (state != cell_state::active) {
+    // Note: We do not need to call dl_harq_buffers.deallocate_ue_buffers(...) because the buffers were reset on cell
+    // deactivation.
+    return launch_no_op_task(false);
+  }
 
-        const du_ue_index_t ue_idx = ue_inst.get_ue_index();
-        const rnti_t        rnti   = ue_inst.get_rnti();
-        if (ue_mng.add_ue(std::move(ue_inst))) {
-          logger.debug("ue={} rnti={} proc=\"MAC UE Creation\": MAC UE DL context created successfully.", ue_idx, rnti);
-          return true;
-        }
-        return false;
-      },
-      [this, ue_index = request.ue_index]() {
-        logger.warning("ue={}: Postponed UE creation. Cause: Task queue is full", fmt::underlying(ue_index));
-      });
+  const du_ue_index_t ue_idx = ue_inst.get_ue_index();
+  const rnti_t        rnti   = ue_inst.get_rnti();
+  if (ue_mng.add_ue(std::move(ue_inst))) {
+    logger.debug("ue={} rnti={} proc=\"MAC UE Creation\": MAC UE DL context created successfully.", ue_idx, rnti);
+    return launch_no_op_task(true);
+  }
+  return launch_no_op_task(false);
 }
 
 async_task<void> mac_cell_processor::remove_ue(const mac_ue_delete_request& request)
@@ -279,18 +260,7 @@ async_task<bool> mac_cell_processor::addmod_bearers(du_ue_index_t
 {
   // Note: logical_channels must outlive the returned async_task completion.
 
-  return defer_and_continue_on_blocking(
-      cell_exec,
-      ctrl_exec,
-      timers,
-      [this, ue_index, logical_channels]() noexcept OCUDU_RTSAN_NONBLOCKING {
-        // Configure logical channels.
-        return state == cell_state::active and ue_mng.addmod_bearers(ue_index, logical_channels);
-      },
-      [this, ue_index]() {
-        logger.warning("ue={}: Postponed UE bearer add/mod operation. Cause: Task queue is full",
-                       fmt::underlying(ue_index));
-      });
+  return launch_no_op_task(state == cell_state::active and ue_mng.addmod_bearers(ue_index, logical_channels));
 }
 
 async_task<bool> mac_cell_processor::remove_bearers(du_ue_index_t ue_index, span<const lcid_t> lcids_to_rem)
