diff --git a/openair2/RRC/NR/MESSAGES/asn1_msg.h b/openair2/RRC/NR/MESSAGES/asn1_msg.h
index 3683a16e64..1d5baa68d9 100644
--- a/openair2/RRC/NR/MESSAGES/asn1_msg.h
+++ b/openair2/RRC/NR/MESSAGES/asn1_msg.h
@@ -117,6 +117,7 @@ int do_nrMeasurementReport_SA(long trigger_to_measid,
                               long rs_type,
                               uint16_t Nid_cell,
                               int rsrp_index,
+                              int sinr_index,
                               int num_neighbor_cells,
                               const uint16_t *neighbor_Nid_cells,
                               const int *neighbor_rsrp_indexes,
