diff --git a/openair2/LAYER2/NR_MAC_gNB/main.c b/openair2/LAYER2/NR_MAC_gNB/main.c
index a8fe8de79c..1f31f543e2 100644
--- a/openair2/LAYER2/NR_MAC_gNB/main.c
+++ b/openair2/LAYER2/NR_MAC_gNB/main.c
@@ -134,6 +134,11 @@ size_t dump_mac_stats(gNB_MAC_INST *gNB, char *output, size_t strlen, bool reset
     }
 
     bool in_sync = !sched_ctrl->ul_failure;
+
+    // Save the previous RSRP values so that querying them is synchronized with the gNB output:
+    stats->prev_num_rsrp_meas = stats->num_rsrp_meas;
+    stats->prev_cumul_rsrp = stats->cumul_rsrp;
+    
     output += snprintf(output,
                        end - output,
                        " %s PH %d dB PCMAX %d dBm, average RSRP %d (%d meas)\n",
