diff --git a/src/xApp/e42_xapp_api.h b/src/xApp/e42_xapp_api.h
index 2e1bbb24..d6696f1a 100644
--- a/src/xApp/e42_xapp_api.h
+++ b/src/xApp/e42_xapp_api.h
@@ -25,7 +25,7 @@ bool try_stop_xapp_api(void);
 
 e2_node_arr_xapp_t e2_nodes_xapp_api(void);
 
-typedef void (*sm_cb)(sm_ag_if_rd_t const*);
+typedef void (*sm_cb)(sm_ag_if_rd_t const*, global_e2_node_id_t const*);
 
 typedef union{
   char* reason;
