diff --git a/examples/xApp/c/monitor/xapp_rc_moni.c b/examples/xApp/c/monitor/xapp_rc_moni.c
index 345892a8..06708c4d 100644
--- a/examples/xApp/c/monitor/xapp_rc_moni.c
+++ b/examples/xApp/c/monitor/xapp_rc_moni.c
@@ -318,7 +318,7 @@ void log_ind_1_2(const e2sm_rc_ind_hdr_frmt_1_t *hdr, const e2sm_rc_ind_msg_frmt
 }
 
 static
-void sm_cb_rc(sm_ag_if_rd_t const* rd)
+void sm_cb_rc(sm_ag_if_rd_t const* rd, global_e2_node_id_t const* node_id)
 {
   assert(rd != NULL);
   assert(rd->type == INDICATION_MSG_AGENT_IF_ANS_V0);
diff --git a/examples/xApp/c/orange/xapp_es_with_cell_util.c b/examples/xApp/c/orange/xapp_es_with_cell_util.c
index 693561b5..c64c0383 100644
--- a/examples/xApp/c/orange/xapp_es_with_cell_util.c
+++ b/examples/xApp/c/orange/xapp_es_with_cell_util.c
@@ -409,7 +409,7 @@ void log_kpm_measurements(kpm_ind_msg_format_1_t const* msg_frm_1)
 }
 
 static
-void sm_cb_kpm(sm_ag_if_rd_t const* rd)
+void sm_cb_kpm(sm_ag_if_rd_t const* rd, global_e2_node_id_t const* node_id)
 {
   assert(rd != NULL);
   assert(rd->type == INDICATION_MSG_AGENT_IF_ANS_V0);
diff --git a/examples/xApp/c/slice/xapp_slice_moni_ctrl.c b/examples/xApp/c/slice/xapp_slice_moni_ctrl.c
index c685b40e..bfacfcfb 100644
--- a/examples/xApp/c/slice/xapp_slice_moni_ctrl.c
+++ b/examples/xApp/c/slice/xapp_slice_moni_ctrl.c
@@ -18,7 +18,7 @@ _Atomic
 uint16_t assoc_rnti = 0;
 
 static
-void sm_cb_slice(sm_ag_if_rd_t const* rd)
+void sm_cb_slice(sm_ag_if_rd_t const* rd, global_e2_node_id_t const* node_id)
 {
   assert(rd != NULL);
   assert(rd->type == INDICATION_MSG_AGENT_IF_ANS_V0);
diff --git a/examples/xApp/c/tc/xapp_tc_all.c b/examples/xApp/c/tc/xapp_tc_all.c
index faf379ed..bd9c7817 100644
--- a/examples/xApp/c/tc/xapp_tc_all.c
+++ b/examples/xApp/c/tc/xapp_tc_all.c
@@ -17,7 +17,7 @@
 
 /*
 static
-void sm_cb_mac(sm_ag_if_rd_t const* rd)
+void sm_cb_mac(sm_ag_if_rd_t const* rd, global_e2_node_id_t const* node_id)
 {
   assert(rd != NULL);
   assert(rd->type == MAC_STATS_V0); 
@@ -31,7 +31,7 @@ static
 int new_queues = 0;
 
 static
-void sm_cb_rlc(sm_ag_if_rd_t const* rd)
+void sm_cb_rlc(sm_ag_if_rd_t const* rd, global_e2_node_id_t const* node_id)
 {
   assert(rd != NULL);
   assert(rd->type == INDICATION_MSG_AGENT_IF_ANS_V0);
@@ -55,7 +55,7 @@ void sm_cb_rlc(sm_ag_if_rd_t const* rd)
 
 /*
 static
-void sm_cb_tc(sm_ag_if_rd_t const* rd)
+void sm_cb_tc(sm_ag_if_rd_t const* rd, global_e2_node_id_t const* node_id)
 {
   assert(rd != NULL);
   assert(rd->type == TC_STATS_V0); 
diff --git a/src/xApp/e42_xapp.c b/src/xApp/e42_xapp.c
index 36fc73f1..0dbfb60f 100644
--- a/src/xApp/e42_xapp.c
+++ b/src/xApp/e42_xapp.c
@@ -195,6 +195,7 @@ e42_xapp_t* init_e42_xapp(fr_args_t const* args)
 
   init_msg_dispatcher(&xapp->msg_disp);
 
+#ifdef SQLITE3_XAPP
   char* dir = get_conf_db_dir(args);
   assert(strlen(dir) < 128 && "String too large");
   char* db_name = get_conf_db_name(args);
@@ -220,6 +221,9 @@ e42_xapp_t* init_e42_xapp(fr_args_t const* args)
 
   free(dir);
   free(db_name);
+#elif defined(NONE_XAPP)
+  init_db_xapp(&xapp->db, NULL);
+#endif
 
 
   const pthread_mutexattr_t *attr = NULL;
diff --git a/src/xApp/e42_xapp_api.h b/src/xApp/e42_xapp_api.h
index 2e1bbb24..d6696f1a 100644
--- a/src/xApp/e42_xapp_api.h
+++ b/src/xApp/e42_xapp_api.h
@@ -25,7 +25,7 @@ bool try_stop_xapp_api(void);
 
 e2_node_arr_xapp_t e2_nodes_xapp_api(void);
 
-typedef void (*sm_cb)(sm_ag_if_rd_t const*);
+typedef void (*sm_cb)(sm_ag_if_rd_t const*, global_e2_node_id_t const*);
 
 typedef union{
   char* reason;
diff --git a/src/xApp/msg_handler_xapp.c b/src/xApp/msg_handler_xapp.c
index 4bb56979..cb666c25 100644
--- a/src/xApp/msg_handler_xapp.c
+++ b/src/xApp/msg_handler_xapp.c
@@ -252,6 +252,7 @@ sm_ind_data_t ind_sm_payload(ric_indication_t const* src)
 
     // Write to the callback. Should I send the E2 Node info to the cb??
     msg_disp.sm_cb = ans.val.sm_cb;
+    msg_disp.e2_node = ans.val.e2_node;
     send_msg_dispatcher(&xapp->msg_disp, &msg_disp );
  }
 
