diff --git a/lib/src/asn1/rrc_nr_utils.cc b/lib/src/asn1/rrc_nr_utils.cc
index d552d890b..5115c75cf 100644
--- a/lib/src/asn1/rrc_nr_utils.cc
+++ b/lib/src/asn1/rrc_nr_utils.cc
@@ -98,6 +98,35 @@ bool make_mac_phr_cfg_t(const phr_cfg_s& asn1_type, phr_cfg_nr_t* phr_cfg_nr)
   return true;
 }
 
+static uint8_t get_nof_cb_preambles_per_ssb(const rach_cfg_common_s& asn1_type)
+{
+  if (!asn1_type.ssb_per_rach_occasion_and_cb_preambs_per_ssb_present) {
+    return 0;
+  }
+
+  const auto& ssb_rach = asn1_type.ssb_per_rach_occasion_and_cb_preambs_per_ssb;
+  switch (ssb_rach.type().value) {
+    case rach_cfg_common_s::ssb_per_rach_occasion_and_cb_preambs_per_ssb_c_::types_opts::one_eighth:
+      return ssb_rach.one_eighth().to_number();
+    case rach_cfg_common_s::ssb_per_rach_occasion_and_cb_preambs_per_ssb_c_::types_opts::one_fourth:
+      return ssb_rach.one_fourth().to_number();
+    case rach_cfg_common_s::ssb_per_rach_occasion_and_cb_preambs_per_ssb_c_::types_opts::one_half:
+      return ssb_rach.one_half().to_number();
+    case rach_cfg_common_s::ssb_per_rach_occasion_and_cb_preambs_per_ssb_c_::types_opts::one:
+      return ssb_rach.one().to_number();
+    case rach_cfg_common_s::ssb_per_rach_occasion_and_cb_preambs_per_ssb_c_::types_opts::two:
+      return ssb_rach.two().to_number();
+    case rach_cfg_common_s::ssb_per_rach_occasion_and_cb_preambs_per_ssb_c_::types_opts::four:
+      return ssb_rach.four();
+    case rach_cfg_common_s::ssb_per_rach_occasion_and_cb_preambs_per_ssb_c_::types_opts::eight:
+      return ssb_rach.eight();
+    case rach_cfg_common_s::ssb_per_rach_occasion_and_cb_preambs_per_ssb_c_::types_opts::sixteen:
+      return ssb_rach.sixteen();
+    default:
+      return 0;
+  }
+}
+
 void make_mac_rach_cfg(const rach_cfg_common_s& asn1_type, rach_cfg_nr_t* rach_cfg_nr)
 {
   rach_cfg_nr->powerRampingStep             = asn1_type.rach_cfg_generic.pwr_ramp_step.to_number();
@@ -106,10 +135,12 @@ void make_mac_rach_cfg(const rach_cfg_common_s& asn1_type, rach_cfg_nr_t* rach_c
   rach_cfg_nr->PreambleReceivedTargetPower  = asn1_type.rach_cfg_generic.preamb_rx_target_pwr;
   rach_cfg_nr->preambleTransMax             = asn1_type.rach_cfg_generic.preamb_trans_max.to_number();
   rach_cfg_nr->ra_ContentionResolutionTimer = asn1_type.ra_contention_resolution_timer.to_number();
+  rach_cfg_nr->nof_preambles                = 64;
 
   if (asn1_type.total_nof_ra_preambs_present) {
     rach_cfg_nr->nof_preambles = asn1_type.total_nof_ra_preambs;
   }
+  rach_cfg_nr->nof_preambles = std::max<uint32_t>(rach_cfg_nr->nof_preambles, get_nof_cb_preambles_per_ssb(asn1_type));
 };
 
 int make_rlc_config_t(const rlc_cfg_c& asn1_type, uint8_t bearer_id, rlc_config_t* cfg_out)
