diff --git a/RAN_Intelligent_Controllers/Near-Real-Time-RIC/full_install.sh b/RAN_Intelligent_Controllers/Near-Real-Time-RIC/full_install.sh
index 286b019..7bec053 100755
--- a/RAN_Intelligent_Controllers/Near-Real-Time-RIC/full_install.sh
+++ b/RAN_Intelligent_Controllers/Near-Real-Time-RIC/full_install.sh
@@ -28,6 +28,9 @@
 # damage to property. The software developed by NIST employees is not subject to
 # copyright protection within the United States.
 
+E2_TERM_PORT=36421 # Default is 36422, which will not modify anything
+E2_TERM_PORT_SUBSTITUTE=36420 # If E2_TERM_PORT is used already, substitute it before replacing with E2_TERM_PORT
+
 # Exit immediately if a command fails
 set -e
 
@@ -139,6 +142,17 @@ else
     # Download ric-dep from gerrit
     if [ ! -d "ric-dep" ]; then
         ./install_scripts/git_clone.sh https://gerrit.o-ran-sc.org/r/ric-plt/ric-dep.git ric-dep
+        # Configure the SCTP E2 termination port (not idempotent so only on first clone)
+        if [ "$E2_TERM_PORT" != "36422" ]; then # Default port
+            echo "Configuring E2 Termination Port to $E2_TERM_PORT..."
+            if sudo find ric-dep/ -type f -exec grep -l "$E2_TERM_PORT_SUBSTITUTE" {} + | grep -q .; then
+                echo "ERROR: The E2 Termination Port Substitute ($E2_TERM_PORT_SUBSTITUTE) is already in use in the following files. Please choose a different substitute port."
+                sudo find ric-dep/ -type f -exec grep -l "$E2_TERM_PORT_SUBSTITUTE" {} +
+                exit 1
+            fi
+            sudo find ric-dep/ -type f -exec sed -i "s/$E2_TERM_PORT/$E2_TERM_PORT_SUBSTITUTE/g" {} +
+            sudo find ric-dep/ -type f -exec sed -i "s/36422/$E2_TERM_PORT/g" {} +
+        fi
     fi
     # Patch the install script and save a backup of the original
     if [ ! -f "ric-dep/bin/install_k8s_and_helm.previous.sh" ]; then
@@ -252,6 +266,17 @@ else
     # Download ric-dep from gerrit
     if [ ! -d "ric-dep" ]; then
         ./install_scripts/git_clone.sh https://gerrit.o-ran-sc.org/r/ric-plt/ric-dep.git ric-dep
+        # Configure the SCTP E2 termination port (not idempotent so only on first clone)
+        if [ "$E2_TERM_PORT" != "36422" ]; then # Default port
+            echo "Configuring E2 Termination Port to $E2_TERM_PORT..."
+            if sudo find ric-dep/ -type f -exec grep -l "$E2_TERM_PORT_SUBSTITUTE" {} + | grep -q .; then
+                echo "ERROR: The E2 Termination Port Substitute ($E2_TERM_PORT_SUBSTITUTE) is already in use in the following files. Please choose a different substitute port."
+                sudo find ric-dep/ -type f -exec grep -l "$E2_TERM_PORT_SUBSTITUTE" {} +
+                exit 1
+            fi
+            sudo find ric-dep/ -type f -exec sed -i "s/$E2_TERM_PORT/$E2_TERM_PORT_SUBSTITUTE/g" {} +
+            sudo find ric-dep/ -type f -exec sed -i "s/36422/$E2_TERM_PORT/g" {} +
+        fi
     fi
 
     echo "Revising RIC Installation YAML File..."
@@ -262,6 +287,7 @@ else
     sudo cp "ric-dep/RECIPE_EXAMPLE/$RIC_YAML_FILE_NAME" "ric-dep/RECIPE_EXAMPLE/$RIC_YAML_FILE_NAME_UPDATED"
     sudo chown $USER:$USER "ric-dep/RECIPE_EXAMPLE/$RIC_YAML_FILE_NAME_UPDATED"
     sudo ./install_scripts/revise_example_recipe_yaml.sh "ric-dep/RECIPE_EXAMPLE/$RIC_YAML_FILE_NAME_UPDATED"
+    sudo ./install_scripts/revise_deployment_for_e2_port.sh 
 
     # Wait for kube-apiserver to be ready before installing Near-RT RIC
     echo "Waiting for the Kubernetes API server to become ready before installing Near-RT RIC..."
@@ -358,6 +384,17 @@ if [ "$(docker ps -aq -f name=^/oransim$ | wc -l)" -ge 1 ] && [ -d "e2-interface
 else
     if [ ! -d "e2-interface" ]; then
         ./install_scripts/git_clone.sh https://gerrit.o-ran-sc.org/r/sim/e2-interface.git
+        # Configure the SCTP E2 termination port (not idempotent so only on first clone)
+        if [ "$E2_TERM_PORT" != "36422" ]; then # Default port
+            echo "Configuring E2 Termination Port to $E2_TERM_PORT..."
+            if sudo find e2-interface/ -type f -exec grep -l "$E2_TERM_PORT_SUBSTITUTE" {} + | grep -q .; then
+                echo "ERROR: The E2 Termination Port Substitute ($E2_TERM_PORT_SUBSTITUTE) is already in use in the following files. Please choose a different substitute port."
+                sudo find e2-interface/ -type f -exec grep -l "$E2_TERM_PORT_SUBSTITUTE" {} +
+                exit 1
+            fi
+            sudo find e2-interface/ -type f -exec sed -i "s/$E2_TERM_PORT/$E2_TERM_PORT_SUBSTITUTE/g" {} +
+            sudo find e2-interface/ -type f -exec sed -i "s/36422/$E2_TERM_PORT/g" {} +
+        fi
     fi
     sudo ./install_scripts/install_e2sim.sh
 fi
diff --git a/RAN_Intelligent_Controllers/Near-Real-Time-RIC/install_scripts/run_e2sim_and_connect_to_ric.sh b/RAN_Intelligent_Controllers/Near-Real-Time-RIC/install_scripts/run_e2sim_and_connect_to_ric.sh
index 3384605..7136859 100755
--- a/RAN_Intelligent_Controllers/Near-Real-Time-RIC/install_scripts/run_e2sim_and_connect_to_ric.sh
+++ b/RAN_Intelligent_Controllers/Near-Real-Time-RIC/install_scripts/run_e2sim_and_connect_to_ric.sh
@@ -33,8 +33,34 @@ echo "# Script: $(realpath "$0")..."
 # Exit immediately if a command fails
 set -e
 
+CURRENT_DIR=$(pwd)
 SCRIPT_DIR=$(dirname "$(realpath "$0")")
-cd "$(dirname "$SCRIPT_DIR")"
+PARENT_DIR=$(dirname "$SCRIPT_DIR")
+cd "$PARENT_DIR"
+
+# Check if docker is accessible from the current user, and if not, repair its permissions
+if [ -z "$FIXED_DOCKER_PERMS" ]; then
+    if ! OUTPUT=$(docker info 2>&1); then
+        if echo "$OUTPUT" | grep -qiE 'permission denied|cannot connect to the docker daemon'; then
+            echo "Docker permissions will repair on reboot."
+            sudo groupadd -f docker
+            if [ -n "$SUDO_USER" ]; then
+                sudo usermod -aG docker "${SUDO_USER:-root}"
+            else
+                sudo usermod -aG docker "${USER:-root}"
+            fi
+            # Rather than requiring a reboot to apply docker permissions, set the docker group and re-run the parent script
+            export FIXED_DOCKER_PERMS=1
+            if ! command -v sg &>/dev/null; then
+                echo
+                echo "WARNING: Could not find set group (sg) command, docker may fail without sudo until the system reboots."
+                echo
+            else
+                exec sg docker -c "$(printf '%q ' "$CURRENT_DIR/$0" "$@")"
+            fi
+        fi
+    fi
+fi
 
 # Set the RAN Function ID if not set
 if [ -z "$RAN_FUNC_ID" ]; then
diff --git a/RAN_Intelligent_Controllers/Near-Real-Time-RIC/install_scripts/stop_e2sim.sh b/RAN_Intelligent_Controllers/Near-Real-Time-RIC/install_scripts/stop_e2sim.sh
index 26d1ae2..72c4077 100755
--- a/RAN_Intelligent_Controllers/Near-Real-Time-RIC/install_scripts/stop_e2sim.sh
+++ b/RAN_Intelligent_Controllers/Near-Real-Time-RIC/install_scripts/stop_e2sim.sh
@@ -33,9 +33,34 @@ echo "# Script: $(realpath "$0")..."
 # Exit immediately if a command fails
 set -e
 
+CURRENT_DIR=$(pwd)
 SCRIPT_DIR=$(dirname "$(realpath "$0")")
 cd "$SCRIPT_DIR"
 
+# Check if docker is accessible from the current user, and if not, repair its permissions
+if [ -z "$FIXED_DOCKER_PERMS" ]; then
+    if ! OUTPUT=$(docker info 2>&1); then
+        if echo "$OUTPUT" | grep -qiE 'permission denied|cannot connect to the docker daemon'; then
+            echo "Docker permissions will repair on reboot."
+            sudo groupadd -f docker
+            if [ -n "$SUDO_USER" ]; then
+                sudo usermod -aG docker "${SUDO_USER:-root}"
+            else
+                sudo usermod -aG docker "${USER:-root}"
+            fi
+            # Rather than requiring a reboot to apply docker permissions, set the docker group and re-run the parent script
+            export FIXED_DOCKER_PERMS=1
+            if ! command -v sg &>/dev/null; then
+                echo
+                echo "WARNING: Could not find set group (sg) command, docker may fail without sudo until the system reboots."
+                echo
+            else
+                exec sg docker -c "$(printf '%q ' "$CURRENT_DIR/$0" "$@")"
+            fi
+        fi
+    fi
+fi
+
 # Stop kpm_sim process inside container if it exists
 if docker exec oransim pgrep -f "kpm_sim" >/dev/null 2>&1; then
     echo "Stopping kpm_sim process..."
