diff --git a/5G_Core_Network/Additional_Cores_5GDeploy/install_scripts/install_docker.sh b/5G_Core_Network/Additional_Cores_5GDeploy/install_scripts/install_docker.sh
index e95317e..1a68da7 100755
--- a/5G_Core_Network/Additional_Cores_5GDeploy/install_scripts/install_docker.sh
+++ b/5G_Core_Network/Additional_Cores_5GDeploy/install_scripts/install_docker.sh
@@ -80,22 +80,32 @@ if ! command -v jq >/dev/null 2>&1; then
 fi
 
 USE_DOCKER_CE=1
-if [ "$USE_DOCKER_CE" -eq 0 ]; then # Use docker.io
-    DOCKERV="29.1"
-
-    # Select a compatible Docker version for Ubuntu 20.*
-    if [[ ${UBUNTU_RELEASE} == 20.* ]]; then
-        DOCKERV="26.1"
-    fi
 
-else # Use docker-ce
-    DOCKERV="29.5"
+# Test Docker version compatibility
+# if [ "$USE_DOCKER_CE" -eq 0 ]; then # Use docker.io
+#     DOCKERV="29.1"
+#
+#     # Select a compatible Docker version for Ubuntu 20.*
+#     if [[ ${UBUNTU_RELEASE} == 20.* ]]; then
+#         DOCKERV="26.1"
+#     fi
+#
+# else # Use docker-ce
+#     DOCKERV="29.5"
+#
+#    # Select a compatible Docker version for Ubuntu 20.*
+#    if [[ ${UBUNTU_RELEASE} == 20.* ]]; then
+#        DOCKERV="28.1"
+#    fi
 
-    # Select a compatible Docker version for Ubuntu 20.*
-    if [[ ${UBUNTU_RELEASE} == 20.* ]]; then
-        DOCKERV="28.1"
+if [ "$USE_DOCKER_CE" -eq 0 ]; then # Use docker.io
+    DOCKERV="20.10"
+    # Select a compatible Docker version for Ubuntu 24.*
+    if [[ ${UBUNTU_RELEASE} == 24.* ]]; then
+        DOCKERV="27.5"
     fi
 
+else # Use docker.ce
     DOCKERV="28.1"
     UBUNTU_CODENAME=$(grep -oP '^UBUNTU_CODENAME=\K.*' /etc/os-release 2>/dev/null)
     # If not found, try to extract VERSION_CODENAME as a fallback
diff --git a/OpenAirInterface_Testbed/Next_Generation_Node_B/install_scripts/install_docker.sh b/OpenAirInterface_Testbed/Next_Generation_Node_B/install_scripts/install_docker.sh
index e95317e..1a68da7 100755
--- a/OpenAirInterface_Testbed/Next_Generation_Node_B/install_scripts/install_docker.sh
+++ b/OpenAirInterface_Testbed/Next_Generation_Node_B/install_scripts/install_docker.sh
@@ -80,22 +80,32 @@ if ! command -v jq >/dev/null 2>&1; then
 fi
 
 USE_DOCKER_CE=1
-if [ "$USE_DOCKER_CE" -eq 0 ]; then # Use docker.io
-    DOCKERV="29.1"
-
-    # Select a compatible Docker version for Ubuntu 20.*
-    if [[ ${UBUNTU_RELEASE} == 20.* ]]; then
-        DOCKERV="26.1"
-    fi
 
-else # Use docker-ce
-    DOCKERV="29.5"
+# Test Docker version compatibility
+# if [ "$USE_DOCKER_CE" -eq 0 ]; then # Use docker.io
+#     DOCKERV="29.1"
+#
+#     # Select a compatible Docker version for Ubuntu 20.*
+#     if [[ ${UBUNTU_RELEASE} == 20.* ]]; then
+#         DOCKERV="26.1"
+#     fi
+#
+# else # Use docker-ce
+#     DOCKERV="29.5"
+#
+#    # Select a compatible Docker version for Ubuntu 20.*
+#    if [[ ${UBUNTU_RELEASE} == 20.* ]]; then
+#        DOCKERV="28.1"
+#    fi
 
-    # Select a compatible Docker version for Ubuntu 20.*
-    if [[ ${UBUNTU_RELEASE} == 20.* ]]; then
-        DOCKERV="28.1"
+if [ "$USE_DOCKER_CE" -eq 0 ]; then # Use docker.io
+    DOCKERV="20.10"
+    # Select a compatible Docker version for Ubuntu 24.*
+    if [[ ${UBUNTU_RELEASE} == 24.* ]]; then
+        DOCKERV="27.5"
     fi
 
+else # Use docker.ce
     DOCKERV="28.1"
     UBUNTU_CODENAME=$(grep -oP '^UBUNTU_CODENAME=\K.*' /etc/os-release 2>/dev/null)
     # If not found, try to extract VERSION_CODENAME as a fallback
diff --git a/RAN_Intelligent_Controllers/Near-Real-Time-RIC/install_patch_files/ric-dep/bin/install_k8s_and_helm.sh b/RAN_Intelligent_Controllers/Near-Real-Time-RIC/install_patch_files/ric-dep/bin/install_k8s_and_helm.sh
index 2553fbd..8960024 100755
--- a/RAN_Intelligent_Controllers/Near-Real-Time-RIC/install_patch_files/ric-dep/bin/install_k8s_and_helm.sh
+++ b/RAN_Intelligent_Controllers/Near-Real-Time-RIC/install_patch_files/ric-dep/bin/install_k8s_and_helm.sh
@@ -160,17 +160,28 @@ USE_DOCKER_CE=1
 if [ "$USE_SYSTEMCTL" = "false" ]; then
     USE_DOCKER_CE=0
 fi
-if [ "$USE_DOCKER_CE" -eq 0 ]; then # Use docker.io
-    DOCKERV="29.1"
 
-    # Select a compatible Docker version for Ubuntu 20.*
-    if [[ ${UBUNTU_RELEASE} == 20.* ]]; then
-        DOCKERV="26.1"
+# Test Docker version compatibility
+# if [ "$USE_DOCKER_CE" -eq 0 ]; then # Use docker.io
+#     DOCKERV="29.1"
+#
+#     # Select a compatible Docker version for Ubuntu 20.*
+#     if [[ ${UBUNTU_RELEASE} == 20.* ]]; then
+#         DOCKERV="26.1"
+#     fi
+#
+# else # Use docker-ce
+#     DOCKERV="29.5"
+
+if [ "$USE_DOCKER_CE" -eq 0 ]; then # Use docker.io
+    DOCKERV="20.10"
+    # Select a compatible Docker version for Ubuntu 24.*
+    if [[ ${UBUNTU_RELEASE} == 24.* ]]; then
+        DOCKERV="27.5"
     fi
 
 else # Use docker-ce
-    DOCKERV="29.5"
-
+    DOCKERV="28.1"
     # Select a compatible Docker version for Ubuntu 20.*
     if [[ ${UBUNTU_RELEASE} == 20.* ]]; then
         DOCKERV="28.1"
diff --git a/RAN_Intelligent_Controllers/Non-Real-Time-RIC/install_patch_files/dep/ric-dep/bin/install_k8s_and_helm.sh b/RAN_Intelligent_Controllers/Non-Real-Time-RIC/install_patch_files/dep/ric-dep/bin/install_k8s_and_helm.sh
index 9059dab..8960024 100755
--- a/RAN_Intelligent_Controllers/Non-Real-Time-RIC/install_patch_files/dep/ric-dep/bin/install_k8s_and_helm.sh
+++ b/RAN_Intelligent_Controllers/Non-Real-Time-RIC/install_patch_files/dep/ric-dep/bin/install_k8s_and_helm.sh
@@ -147,7 +147,7 @@ sudo env $APTVARS apt-get install -y jq netcat-openbsd make ipset moreutils
 # The version will be dynamically completed rather than hardcoding in the version
 KUBEV="1.33"
 KUBECNIV="1.6"
-HELMV="4.2"
+HELMV="3.20"
 
 # Fetch the Ubuntu release version regardless of the derivative distro
 if [ -f /etc/upstream-release/lsb-release ]; then
@@ -160,17 +160,28 @@ USE_DOCKER_CE=1
 if [ "$USE_SYSTEMCTL" = "false" ]; then
     USE_DOCKER_CE=0
 fi
-if [ "$USE_DOCKER_CE" -eq 0 ]; then # Use docker.io
-    DOCKERV="29.1"
 
-    # Select a compatible Docker version for Ubuntu 20.*
-    if [[ ${UBUNTU_RELEASE} == 20.* ]]; then
-        DOCKERV="26.1"
+# Test Docker version compatibility
+# if [ "$USE_DOCKER_CE" -eq 0 ]; then # Use docker.io
+#     DOCKERV="29.1"
+#
+#     # Select a compatible Docker version for Ubuntu 20.*
+#     if [[ ${UBUNTU_RELEASE} == 20.* ]]; then
+#         DOCKERV="26.1"
+#     fi
+#
+# else # Use docker-ce
+#     DOCKERV="29.5"
+
+if [ "$USE_DOCKER_CE" -eq 0 ]; then # Use docker.io
+    DOCKERV="20.10"
+    # Select a compatible Docker version for Ubuntu 24.*
+    if [[ ${UBUNTU_RELEASE} == 24.* ]]; then
+        DOCKERV="27.5"
     fi
 
 else # Use docker-ce
-    DOCKERV="29.5"
-
+    DOCKERV="28.1"
     # Select a compatible Docker version for Ubuntu 20.*
     if [[ ${UBUNTU_RELEASE} == 20.* ]]; then
         DOCKERV="28.1"
