diff --git a/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm_subs.c b/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm_subs.c
index d3381cadd3..4684a69959 100644
--- a/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm_subs.c
+++ b/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm_subs.c
@@ -22,6 +22,7 @@
 #include "ran_func_kpm_subs.h"
 
 #include <search.h>
+#include <stdlib.h>
 
 e2_node_level_stats_t cp_node_level_stats(const e2_node_level_stats_t *src)
 {
@@ -32,6 +33,9 @@ e2_node_level_stats_t cp_node_level_stats(const e2_node_level_stats_t *src)
     .mac_stats.ul.used_prb_aggregate = src->mac_stats.ul.used_prb_aggregate,
   };
 
+  // Copy distribution bins
+  memcpy(dst.mac_stats.rsrp_dist_bins, src->mac_stats.rsrp_dist_bins, sizeof(dst.mac_stats.rsrp_dist_bins));
+
   return dst;
 }
 
@@ -201,6 +205,28 @@ static meas_record_lst_t fill_RRU_PrbTotUl(__attribute__((unused))uint32_t gran_
 }
 #endif
 
+// Added metric for research purposes only
+static meas_record_lst_t fill_L1M_SS_RSRP_BIN(__attribute__((unused))uint32_t gran_period_ms, cudu_ue_info_pair_t ue_info, __attribute__((unused))const size_t ue_idx, __attribute__((unused))e2_node_level_stats_t* node_stats)
+{
+  meas_record_lst_t meas_record = {0};
+
+  const size_t num_bins = 128; // Distribution for RSRP 0-127
+  meas_record.value = DIST_MEAS_VALUE;
+  meas_record.dist.len = num_bins;
+  meas_record.dist.val = calloc(num_bins, sizeof(uint64_t)); 
+
+  if (meas_record.dist.val != NULL) {
+    uint64_t *bins = (uint64_t*)meas_record.dist.val;
+
+    for (size_t i = 0; i < num_bins; i++) {
+        // Calculate delta: current (node_stats[1]) - previous (node_stats[0])
+        bins[i] = node_stats[1].mac_stats.rsrp_dist_bins[i] - node_stats[0].mac_stats.rsrp_dist_bins[i];
+    }
+  }
+  
+  return meas_record;
+}
+
 static kv_measure_t lst_measure[] = {
   {.key = "DRB.PdcpSduVolumeDL", .value = fill_DRB_PdcpSduVolumeDL }, 
   {.key = "DRB.PdcpSduVolumeUL", .value = fill_DRB_PdcpSduVolumeUL },
@@ -211,6 +237,7 @@ static kv_measure_t lst_measure[] = {
   {.key = "RRU.PrbTotDl", .value =  fill_RRU_PrbTotDl }, 
   {.key = "RRU.PrbTotUl", .value =  fill_RRU_PrbTotUl }, 
 #endif
+  {.key = "L1M.SS-RSRP.Bin", .value = fill_L1M_SS_RSRP_BIN },
 }; 
 
 void init_kpm_subs_data(void)
