diff --git a/radio/zmq/ring_buffer.cpp b/radio/zmq/ring_buffer.cpp
index 9aa89a5125..a7aa8aaa20 100644
--- a/radio/zmq/ring_buffer.cpp
+++ b/radio/zmq/ring_buffer.cpp
@@ -10,7 +10,7 @@
 template <typename T>
 ring_buffer<T>::ring_buffer(size_t max_size) : max_size_(max_size)
 {
-  buffer_ = std::make_unique<T[]>(max_size);
+  buffer_ = std::unique_ptr<T[]>(new T[max_size]());
 }
 
 template <typename T>
