--- a/radio/zmq/zmq_imported.h
+++ b/radio/zmq/zmq_imported.h
@@ -19,8 +19,8 @@
  public:
   void *socket_;
   overflow_buffer buffer_;
-  std::atomic<uint64_t> sample_count_ = 0;
-  std::atomic<bool> is_tx_enabled_ = false;
+  std::atomic<uint64_t> sample_count_{0};
+  std::atomic<bool> is_tx_enabled_{false};
   std::mutex transmit_alignment_mutex_;
   std::condition_variable transmit_alignment_cvar_;
 
