diff --git a/radio/zmq/zmq_imported.cpp b/radio/zmq/zmq_imported.cpp
index 51dd842609..6008abac7f 100644
--- a/radio/zmq/zmq_imported.cpp
+++ b/radio/zmq/zmq_imported.cpp
@@ -14,7 +14,7 @@ static constexpr std::chrono::milliseconds RECEIVE_TS_ALIGN_TIMEOUT = std::chron
 
 void zmq_tx_channel::transmit(c16_t *samples, size_t nsamps, uint64_t timestamp)
 {
-  std::scoped_lock lock(transmit_alignment_mutex_);
+  std::lock_guard<std::mutex> lock(transmit_alignment_mutex_);
 
   size_t zeros_to_push = 0;
   if (timestamp > sample_count_) {
