diff --git a/srsue/src/phy/nr/cc_worker.cc b/srsue/src/phy/nr/cc_worker.cc
index 46e059054..d252e7a98 100644
--- a/srsue/src/phy/nr/cc_worker.cc
+++ b/srsue/src/phy/nr/cc_worker.cc
@@ -254,8 +254,8 @@ bool cc_worker::decode_pdsch_dl()
 
   // Abort if MAC says it doesn't need the TB
   if (not dl_action.tb.enabled) {
-    // Force positive ACK
-    if (pdsch_cfg.grant.rnti_type == srsran_rnti_type_c) {
+    // Force positive HARQ-ACK
+    if (pdsch_cfg.grant.rnti_type == srsran_rnti_type_c || pdsch_cfg.grant.rnti_type == srsran_rnti_type_tc) {
       phy.set_pending_ack(dl_slot_cfg.idx, ack_resource, true);
     }
 
@@ -335,8 +335,8 @@ bool cc_worker::decode_pdsch_dl()
     }
   }
 
-  // Enqueue PDSCH ACK information only if the RNTI is type C
-  if (pdsch_cfg.grant.rnti_type == srsran_rnti_type_c) {
+  // Enqueue HARQ-ACK for PDSCH scheduled by C-RNTI or TC-RNTI.
+  if (pdsch_cfg.grant.rnti_type == srsran_rnti_type_c || pdsch_cfg.grant.rnti_type == srsran_rnti_type_tc) {
     phy.set_pending_ack(dl_slot_cfg.idx, ack_resource, pdsch_res.tb[0].crc);
   }
 
