diff --git a/cmake_targets/tools/build_helper b/cmake_targets/tools/build_helper
index 3fc1db7b56..9f68588ea6 100755
--- a/cmake_targets/tools/build_helper
+++ b/cmake_targets/tools/build_helper
@@ -37,6 +37,7 @@ case "$OS_DISTRO" in
   centos) OS_BASEDISTRO="fedora"; INSTALLER="dnf"; CMAKE="cmake3" ;; # CentOS Stream
   debian) OS_BASEDISTRO="debian"; INSTALLER="apt-get"; CMAKE="cmake" ;;
   ubuntu) OS_BASEDISTRO="debian"; INSTALLER="apt-get"; CMAKE="cmake" ;;
+  linuxmint) OS_BASEDISTRO="debian"; INSTALLER="apt-get"; CMAKE="cmake" ;;
 esac
 KERNEL_VERSION=$(uname -r | cut -d '.' -f1)
 KERNEL_MAJOR=$(uname -r | cut -d '.' -f2)
@@ -93,6 +94,18 @@ echo_info()    { cecho "$*" $blue         ;}
 # Examples:
 #   debian8.5
 get_distribution_release() {
+    if [[ $OS_DISTRO == "linuxmint" ]]; then
+        if [[ $OS_RELEASE == 22* ]]; then
+            OS_DISTRO="ubuntu"
+            OS_RELEASE="24.04"
+        elif [[ $OS_RELEASE == 22* ]]; then
+            OS_DISTRO="ubuntu"
+            OS_RELEASE="22.04"
+        elif [[ $OS_RELEASE == 20* ]]; then
+            OS_DISTRO="ubuntu"
+            OS_RELEASE="20.04"
+        fi
+    fi
     if [[ ! -z "$OS_DISTRO$OS_RELEASE" ]]; then
         echo "$OS_DISTRO$OS_RELEASE"
     else
