diff --git a/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm.c b/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm.c
index 7a96cb1f4a..d434786e04 100644
--- a/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm.c
+++ b/openair2/E2AP/RAN_FUNCTION/O-RAN/ran_func_kpm.c
@@ -83,6 +83,7 @@ static meas_data_lst_t fill_kpm_meas_data_item(const meas_info_format_1_lst_t* m
 static kpm_ind_msg_format_1_t fill_kpm_ind_msg_frm_1(cudu_ue_info_pair_t ue_info, const size_t ue_idx, const kpm_act_def_format_1_t* act_def_fr_1)
 {
   kpm_ind_msg_format_1_t msg_frm_1 = {0};
+  ue_info.cell_global_id = act_def_fr_1->cell_global_id;
 
   // Measurement Data contains a set of Meas Records, each collected at each granularity period
   msg_frm_1.meas_data_lst_len = 1;  /*  this value is equal to (kpm_ric_event_trigger_format_1.report_period_ms/act_def_fr_1->gran_period_ms)
@@ -441,6 +442,7 @@ bool read_kpm_sm(void* data)
 static const char* kpm_node_meas_du[] = {
   "CARR.PDSCHMCSDist",
   "CARR.PUSCHMCSDist",
+  "CARR.WBCQIDist",
   "L1M.SS-RSRP",
   NULL,
 };
@@ -454,6 +456,7 @@ static const char* kpm_node_meas_cu[] = {
 static const char* kpm_node_meas_gnb[] = {
   "CARR.PDSCHMCSDist",
   "CARR.PUSCHMCSDist",
+  "CARR.WBCQIDist",
   "L1M.SS-RSRP",
   "MR.NRScSSSINR",
   "RRC.ConnMean",
@@ -466,6 +469,23 @@ static const char* kpm_meas_du[] = {
   "DRB.UEThpUl",
   "RRU.PrbTotDl",
   "RRU.PrbTotUl",
+
+  "PHY.NPrbDl",
+  "PHY.DerivedRssiDl",
+  "PHY.DerivedRsrqDl",
+  "PUSCH.Snr",
+  "PUCCH.Snr",
+  "DRB.HarqMcsUl",
+  "DRB.HarqMcsDl",
+  "DRB.HarqBlockErrorRateUl",
+  "DRB.HarqBlockErrorRateDl",
+  "DRB.MacSduRetransmissionRateUl",
+  "DRB.MacSduRetransmissionRateDl",
+  "DRB.MacSduErrorRateUl",
+  "DRB.MacSduErrorRateDl",
+  "DRB.DerivedCQIDl",
+  "PHY.CqiWb1TbDl",
+  "PHY.CqiWb2TbDl",
   NULL,
 };
 
@@ -477,6 +497,23 @@ static const char* kpm_meas_gnb[] = {
   "DRB.UEThpUl",
   "RRU.PrbTotDl",
   "RRU.PrbTotUl",
+
+  "PHY.NPrbDl",
+  "PHY.DerivedRssiDl",
+  "PHY.DerivedRsrqDl",
+  "PUSCH.Snr",
+  "PUCCH.Snr",
+  "DRB.HarqMcsUl",
+  "DRB.HarqMcsDl",
+  "DRB.HarqBlockErrorRateUl",
+  "DRB.HarqBlockErrorRateDl",
+  "DRB.MacSduRetransmissionRateUl",
+  "DRB.MacSduRetransmissionRateDl",
+  "DRB.MacSduErrorRateUl",
+  "DRB.MacSduErrorRateDl",
+  "DRB.DerivedCQIDl",
+  "PHY.CqiWb1TbDl",
+  "PHY.CqiWb2TbDl",
   NULL,
 };
 
