diff --git a/lib/rlc/rlc_tx_tm_entity.h b/lib/rlc/rlc_tx_tm_entity.h
index a7132a3cb2..eba8d4c2bb 100644
--- a/lib/rlc/rlc_tx_tm_entity.h
+++ b/lib/rlc/rlc_tx_tm_entity.h
@@ -66,16 +66,14 @@ public:
 
 private:
   /// Called whenever the buffer state has been changed by upper layers (new SDUs or SDU discard) so that lower layers
-  /// need to be informed about the new buffer state. This function defers the actual notification \c
-  /// handle_changed_buffer_state to pcell_executor. The notification is discarded if another notification is already
-  /// queued for execution. This function should not be called from \c pull_pdu, since the lower layer accounts for the
-  /// amount of extracted data itself.
+  /// need to be informed about the new buffer state. The notification is discarded if another notification is already
+  /// being processed. This function should not be called from \c pull_pdu, since the lower layer accounts for the amount
+  /// of extracted data itself.
   ///
   /// Safe execution from: Any executor
   void handle_changed_buffer_state();
 
-  /// Immediately informs the lower layer of the current buffer state. This function is called from pcell_executor and
-  /// its execution is queued by \c handle_changed_buffer_state.
+  /// Immediately informs the lower layer of the current buffer state.
   ///
   /// Safe execution from: pcell_executor
   void update_mac_buffer_state() noexcept;
