diff --git a/lib/rlc/rlc_tx_am_entity.h b/lib/rlc/rlc_tx_am_entity.h
index 93374cd32c..d855703583 100644
--- a/lib/rlc/rlc_tx_am_entity.h
+++ b/lib/rlc/rlc_tx_am_entity.h
@@ -346,19 +346,17 @@ private:
   void check_sn_reached_max_retx(uint32_t sn);
 
   /// Called whenever the buffer state has been changed by upper layers (i.e new SDU or SDU discard) or the size of the
-  /// own status PDU has changed so that lower layers need to be informed about the new buffer state. This function
-  /// defers the actual notification \c handle_changed_buffer_state to pcell_executor. The notification is discarded if
-  /// another notification is already queued for execution. This function should not be called from \c pull_pdu, since
-  /// the lower layer accounts for the amount of extracted data itself.
+  /// own status PDU has changed so that lower layers need to be informed about the new buffer state. The notification
+  /// is discarded if another notification is already being processed. This function should not be called from \c
+  /// pull_pdu, since the lower layer accounts for the amount of extracted data itself.
   ///
   /// Safe execution from: Any executor
   void handle_changed_buffer_state();
 
   /// Immediately informs the lower layer of the current buffer state. This function must be called from pcell_executor.
   /// It is used for buffer update notifications by non-excessive internal RLC events (i.e. expired poll retransmit
-  /// timer or handled received status report), and for defered handling of \c handle_changed_buffer_state. This
-  /// function should not be called from \c pull_pdu, since the lower layer accounts for the amount of extracted data
-  /// itself.
+  /// timer or handled received status report). This function should not be called from \c pull_pdu, since the lower
+  /// layer accounts for the amount of extracted data itself.
   ///
   /// Safe execution from: pcell_executor
   /// \param force_notify forces a notification of the lower layer regardless of the current/previous buffer state.
