diff --git a/openair2/LAYER2/NR_MAC_gNB/gNB_scheduler_uci.c b/openair2/LAYER2/NR_MAC_gNB/gNB_scheduler_uci.c
index b779489abd..2e83b43034 100644
--- a/openair2/LAYER2/NR_MAC_gNB/gNB_scheduler_uci.c
+++ b/openair2/LAYER2/NR_MAC_gNB/gNB_scheduler_uci.c
@@ -564,6 +564,10 @@ static void evaluate_rsrp_report(gNB_MAC_INST *nrmac,
     return;
   }
 
+  // cell-wide SS-RSRP distribution, see 28.552 5.1.1.22.1
+  NR_du_stats_t *du_stats = &RC.nrmac[0]->du_stats;
+  du_stats->ss_rsrp_dist[rsrp] += 1;
+
   for (RSRP_report_t *i = rsrp_report->r + 1; i < rsrp_report->r + rsrp_report->nb; i++) {
     curr_payload = pickandreverse_bits(payload, 4, *cumul_bits);
     csi_report->nb_of_csi_ssb_report++;
@@ -573,6 +577,7 @@ static void evaluate_rsrp_report(gNB_MAC_INST *nrmac,
   }
 
   NR_mac_stats_t *stats = &UE->mac_stats;
+
   // including ssb rsrp in mac stats
   stats->cumul_rsrp += rsrp_report->r[0].RSRP;
   stats->num_rsrp_meas++;
