diff --git a/openair2/LAYER2/NR_MAC_gNB/gNB_scheduler_uci.c b/openair2/LAYER2/NR_MAC_gNB/gNB_scheduler_uci.c
index 682ea765ac..8595dc3a09 100644
--- a/openair2/LAYER2/NR_MAC_gNB/gNB_scheduler_uci.c
+++ b/openair2/LAYER2/NR_MAC_gNB/gNB_scheduler_uci.c
@@ -615,7 +615,12 @@ static void evaluate_rsrp_report(gNB_MAC_INST *nrmac,
   }
 
   NR_mac_stats_t *stats = &UE->mac_stats;
-  // including ssb rsrp in mac stats
+
+  // log each RSRP measurements for E2 reporting (resets after each E2 report, periodicity specified over E2SM)
+  stats->e2_cumul_rsrp += rsrp_report->RSRP[0];
+  stats->e2_num_rsrp_meas++;
+
+  // including ssb rsrp in mac stats (resets after printed on gNB, every 1280 ms)
   stats->cumul_rsrp += rsrp_report->RSRP[0];
   stats->num_rsrp_meas++;
 }
