diff --git a/openair2/LAYER2/NR_MAC_gNB/gNB_scheduler_dlsch.c b/openair2/LAYER2/NR_MAC_gNB/gNB_scheduler_dlsch.c
index b69490fbc9..f36b111a6e 100644
--- a/openair2/LAYER2/NR_MAC_gNB/gNB_scheduler_dlsch.c
+++ b/openair2/LAYER2/NR_MAC_gNB/gNB_scheduler_dlsch.c
@@ -379,6 +379,9 @@ void abort_nr_dl_harq(NR_UE_info_t* UE, int8_t harq_pid)
 
   finish_nr_dl_harq(sched_ctrl, harq_pid);
   UE->mac_stats.dl.errors++;
+
+  NR_UE_harq_t *harq = &sched_ctrl->harq_processes[harq_pid];
+  UE->mac_stats.dl.sdu_errors += harq->num_sdus; // SDU-level failures
 }
 
 typedef struct {
@@ -1370,6 +1373,7 @@ void nr_schedule_ue_spec(module_id_t module_id,
       gNB_mac->mac_stats.used_prb_aggregate += sched_pdsch->rbSize;
 
       /* save retransmission information */
+      harq->num_sdus = sdus;
       harq->sched_pdsch = *sched_pdsch;
       /* save which time allocation has been used, to be used on
        * retransmissions */
