diff --git a/openair2/LAYER2/NR_MAC_gNB/main.c b/openair2/LAYER2/NR_MAC_gNB/main.c
index 3137ce0902..55ac746863 100644
--- a/openair2/LAYER2/NR_MAC_gNB/main.c
+++ b/openair2/LAYER2/NR_MAC_gNB/main.c
@@ -135,6 +135,7 @@ size_t dump_mac_stats(gNB_MAC_INST *gNB, char *output, size_t strlen, bool reset
     }
 
     bool in_sync = !sched_ctrl->ul_failure;
+    
     output += snprintf(output,
                        end - output,
                        " %s PH %d dB PCMAX %d dBm, average RSRP %d (%d meas)\n",
