diff --git a/openair2/LAYER2/NR_MAC_UE/tests/test_nr_ue_ra_procedures.cpp b/openair2/LAYER2/NR_MAC_UE/tests/test_nr_ue_ra_procedures.cpp
index 54f0a77933..8d542f1a6c 100644
--- a/openair2/LAYER2/NR_MAC_UE/tests/test_nr_ue_ra_procedures.cpp
+++ b/openair2/LAYER2/NR_MAC_UE/tests/test_nr_ue_ra_procedures.cpp
@@ -135,7 +135,8 @@ void nr_mac_rrc_meas_ind_ue(module_id_t module_id,
                             uint16_t Nid_cell,
                             bool csi_meas,
                             bool is_neighboring_cell,
-                            int rsrp_dBm)
+                            int rsrp_dBm,
+                            int sinr_index)
 {
   UNUSED(module_id);
   UNUSED(gNB_index);
@@ -143,6 +144,7 @@ void nr_mac_rrc_meas_ind_ue(module_id_t module_id,
   UNUSED(csi_meas);
   UNUSED(is_neighboring_cell);
   UNUSED(rsrp_dBm);
+  UNUSED(sinr_index);
 }
 }
 #include <cstdio>
