diff --git a/src/sm/kpm_sm/kpm_sm_v02.01/dec/dec_asn/dec_ric_ind_hdr_frm_1.c b/src/sm/kpm_sm/kpm_sm_v02.01/dec/dec_asn/dec_ric_ind_hdr_frm_1.c
index e2bfc058..c8aedc0f 100644
--- a/src/sm/kpm_sm/kpm_sm_v02.01/dec/dec_asn/dec_ric_ind_hdr_frm_1.c
+++ b/src/sm/kpm_sm/kpm_sm_v02.01/dec/dec_asn/dec_ric_ind_hdr_frm_1.c
@@ -2,18 +2,42 @@
  * SPDX-License-Identifier: LicenseRef-CSSL-1.0
  */
 
-#include "../../../../../util/conversions.h"
 #include <assert.h>
-#include <arpa/inet.h>
+#include "../../../../../util/conversions.h"
+#include <arpa/inet.h> // ntohl
+
 #include "dec_ric_ind_hdr_frm_1.h"
 
+// #define ntohll(x) ((1==ntohl(1)) ? (x) : ((uint64_t)ntohl((x) & 0xFFFFFFFF) << 32) | ntohl((x) >> 32))
+
+// uint64_t ntohll(uint64_t x)
+// {
+//   return (uint64_t)((uint64_t)ntohl((x) & 0xFFFFFFFFLL) << 32) | (uint32_t)ntohl(x);
+// }
+
+// #define ntohll(x) ((((uint64_t)ntohl(x)) << 32) + ntohl((x) >> 32))
+
+// #define NTOHLL(x) ((1==ntohl(1)) ? (x) : (((uint64_t)ntohl((x) & 0xFFFFFFFFUL)) << 32) | ntohl((uint32_t)((x) >> 32)))
+
+uint64_t ntohll(uint64_t x)
+{
+  uint32_t x1 = (uint32_t)((x & 0xFFFFFFFF00000000LL) >> 32);
+  uint32_t x2 = (uint32_t)(x & 0xFFFFFFFFLL);
+  x1 = ntohl(x1);
+  x2 = ntohl(x2);
+  uint64_t v = ((uint64_t)x1) << 32 | x2;
+
+  return v;
+}
+
+
 kpm_ric_ind_hdr_format_1_t kpm_dec_ind_hdr_frm_1_asn(const E2SM_KPM_IndicationHeader_Format1_t * ind_hdr_asn)
 {
     kpm_ric_ind_hdr_format_1_t ind_hdr = {0};
 
-    uint32_t reversed_ts;
-    OCTET_STRING_TO_INT32(&ind_hdr_asn->colletStartTime, reversed_ts);
-    ind_hdr.collectStartTime = ntohl(reversed_ts);
+    uint64_t reversed_ts;
+    reversed_ts = octet_string_to_int_64(ind_hdr_asn->colletStartTime);
+    ind_hdr.collectStartTime = ntohll(reversed_ts);
     
     if (ind_hdr_asn->fileFormatversion != NULL){
       ind_hdr.fileformat_version = calloc(1, sizeof(byte_array_t));
