diff --git a/OpenAirInterface_Testbed/RAN_Intelligent_Controllers/Flexible-RIC/full_install.sh b/OpenAirInterface_Testbed/RAN_Intelligent_Controllers/Flexible-RIC/full_install.sh
index 18afbc0..b12aa7e 100755
--- a/OpenAirInterface_Testbed/RAN_Intelligent_Controllers/Flexible-RIC/full_install.sh
+++ b/OpenAirInterface_Testbed/RAN_Intelligent_Controllers/Flexible-RIC/full_install.sh
@@ -65,10 +65,10 @@ fi
 INSTALL_START_TIME=$(date +%s)
 
 echo "Installing dependencies..."
-if ! command -v gcc-10 &>/dev/null || ! command -v g++-10 &>/dev/null || ! command -v swig &>/dev/null; then
+if ! command -v gcc &>/dev/null || ! command -v g++ &>/dev/null || ! command -v swig &>/dev/null; then
     sudo apt-get update || true
     sudo env $APTVARS apt-get install -y build-essential automake
-    sudo env $APTVARS apt-get install -y gcc-10 g++-10
+    sudo env $APTVARS apt-get install -y gcc g++
     sudo env $APTVARS apt-get install -y libsctp-dev python3 cmake-curses-gui libpcre2-dev python3-dev
 fi
 
@@ -131,7 +131,7 @@ cd flexric
 sudo rm -rf build
 mkdir build
 cd build
-CC=gcc-10 CXX=g++-10 cmake .. -DE2AP_VERSION=$E2AP_VERSION -DKPM_VERSION=$KPM_VERSION $ADDITIONAL_FLAGS
+CC=gcc CXX=g++ cmake .. -DE2AP_VERSION=$E2AP_VERSION -DKPM_VERSION=$KPM_VERSION $ADDITIONAL_FLAGS
 make -j$(nproc)
 
 echo "Installing FlexRIC..."
