diff --git a/openair2/RRC/NR_UE/L2_interface_ue.c b/openair2/RRC/NR_UE/L2_interface_ue.c
index 7a426fb2b1..b80f16d42e 100644
--- a/openair2/RRC/NR_UE/L2_interface_ue.c
+++ b/openair2/RRC/NR_UE/L2_interface_ue.c
@@ -13,7 +13,13 @@
 
 typedef uint32_t channel_t;
 
-void nr_mac_rrc_meas_ind_ue(module_id_t module_id, uint32_t gNB_index, uint16_t Nid_cell, bool csi_meas, bool is_neighboring_cell, int rsrp_dBm)
+void nr_mac_rrc_meas_ind_ue(module_id_t module_id,
+                            uint32_t gNB_index,
+                            uint16_t Nid_cell,
+                            bool csi_meas,
+                            bool is_neighboring_cell,
+                            int rsrp_dBm,
+                            int sinr_index)
 {
   MessageDef *message_p = itti_alloc_new_message(TASK_MAC_UE, 0, NR_RRC_MAC_MEAS_DATA_IND);
   NR_RRC_MAC_MEAS_DATA_IND(message_p).gnb_index = gNB_index;
@@ -21,6 +27,7 @@ void nr_mac_rrc_meas_ind_ue(module_id_t module_id, uint32_t gNB_index, uint16_t
   NR_RRC_MAC_MEAS_DATA_IND(message_p).is_csi_meas = csi_meas;
   NR_RRC_MAC_MEAS_DATA_IND(message_p).is_neighboring_cell = is_neighboring_cell;
   NR_RRC_MAC_MEAS_DATA_IND(message_p).rsrp_dBm = rsrp_dBm;
+  NR_RRC_MAC_MEAS_DATA_IND(message_p).sinr_index = sinr_index;
   itti_send_msg_to_task(TASK_RRC_NRUE, GNB_MODULE_ID_TO_INSTANCE(module_id), message_p);
 }
 
