diff --git a/openair2/RRC/NR_UE/rrc_defs.h b/openair2/RRC/NR_UE/rrc_defs.h
index 81e23b43ca..123ece80c1 100644
--- a/openair2/RRC/NR_UE/rrc_defs.h
+++ b/openair2/RRC/NR_UE/rrc_defs.h
@@ -185,6 +185,12 @@ typedef struct l3_measurements_s {
   NR_timer_t TA2;
   NR_timer_t TA3;
   NR_timer_t periodic_report_timer;
+  long periodical_trigger_to_measid;
+  long periodical_rs_type;
+  int periodical_reports_sent;
+  int periodical_max_reports;
+  long periodical_report_interval_ms;
+  NR_timer_t periodical_report_timer;
 } l3_measurements_t;
 
 typedef struct rrcPerNB {
