diff --git a/openair2/RRC/NR_UE/rrc_proto.h b/openair2/RRC/NR_UE/rrc_proto.h
index f4bf1b4e1e..9bf44a05fa 100644
--- a/openair2/RRC/NR_UE/rrc_proto.h
+++ b/openair2/RRC/NR_UE/rrc_proto.h
@@ -28,7 +28,12 @@ void nr_rrc_going_to_IDLE(NR_UE_RRC_INST_t *rrc,
 
 void handle_RRCRelease(NR_UE_RRC_INST_t *rrc);
 
-void rrc_ue_generate_measurementReport(rrcPerNB_t *rrc, instance_t ue_id);
+void rrc_ue_generate_measurementReport(rrcPerNB_t *rrc,
+                                       instance_t ue_id,
+                                       long meas_id,
+                                       long trigger_quantity,
+                                       long rs_type,
+                                       bool neighbor_cell_valid);
 
 void set_rlf_sib1_timers_and_constants(NR_UE_Timers_Constants_t *tac, NR_UE_TimersAndConstants_t *ue_TimersAndConstants);
 
