diff --git a/openair2/RRC/NR_UE/rrc_timers_and_constants.c b/openair2/RRC/NR_UE/rrc_timers_and_constants.c
index f912d6b8c6..ffb92b9b7d 100644
--- a/openair2/RRC/NR_UE/rrc_timers_and_constants.c
+++ b/openair2/RRC/NR_UE/rrc_timers_and_constants.c
@@ -114,7 +114,12 @@ void handle_meas_timers(NR_UE_RRC_INST_t *rrc)
     bool ta2_expired = nr_timer_tick(&l3_measurements->TA2);
     bool ta3_expired = nr_timer_tick(&l3_measurements->TA3);
     if (ta2_expired && l3_measurements->trigger_quantity > 0) {
-      rrc_ue_generate_measurementReport(nb, rrc->ue_id);
+      rrc_ue_generate_measurementReport(nb,
+                                        rrc->ue_id,
+                                        l3_measurements->trigger_to_measid,
+                                        l3_measurements->trigger_quantity,
+                                        l3_measurements->rs_type,
+                                        l3_measurements->neighbor_cell_valid);
       l3_measurements->reports_sent = 1;
 
       if (l3_measurements->reports_sent < l3_measurements->max_reports
@@ -125,7 +130,12 @@ void handle_meas_timers(NR_UE_RRC_INST_t *rrc)
     }
 
     if (ta3_expired && l3_measurements->trigger_quantity > 0) {
-      rrc_ue_generate_measurementReport(nb, rrc->ue_id);
+      rrc_ue_generate_measurementReport(nb,
+                                        rrc->ue_id,
+                                        l3_measurements->trigger_to_measid,
+                                        l3_measurements->trigger_quantity,
+                                        l3_measurements->rs_type,
+                                        l3_measurements->neighbor_cell_valid);
       l3_measurements->reports_sent = 1;
 
       if (l3_measurements->reports_sent < l3_measurements->max_reports
@@ -137,7 +147,12 @@ void handle_meas_timers(NR_UE_RRC_INST_t *rrc)
 
     bool periodic_expired = nr_timer_tick(&l3_measurements->periodic_report_timer);
     if (periodic_expired && l3_measurements->reports_sent < l3_measurements->max_reports) {
-      rrc_ue_generate_measurementReport(nb, rrc->ue_id);
+      rrc_ue_generate_measurementReport(nb,
+                                        rrc->ue_id,
+                                        l3_measurements->trigger_to_measid,
+                                        l3_measurements->trigger_quantity,
+                                        l3_measurements->rs_type,
+                                        l3_measurements->neighbor_cell_valid);
       l3_measurements->reports_sent++;
 
       if (l3_measurements->reports_sent < l3_measurements->max_reports) {
@@ -145,6 +160,25 @@ void handle_meas_timers(NR_UE_RRC_INST_t *rrc)
         nr_timer_start(&l3_measurements->periodic_report_timer);
       }
     }
+
+    bool periodical_expired = nr_timer_tick(&l3_measurements->periodical_report_timer);
+    if (periodical_expired
+        && l3_measurements->periodical_reports_sent < l3_measurements->periodical_max_reports) {
+      rrc_ue_generate_measurementReport(nb,
+                                        rrc->ue_id,
+                                        l3_measurements->periodical_trigger_to_measid,
+                                        NR_MeasTriggerQuantityOffset_PR_rsrp,
+                                        l3_measurements->periodical_rs_type,
+                                        false);
+      l3_measurements->periodical_reports_sent++;
+
+      if (l3_measurements->periodical_reports_sent < l3_measurements->periodical_max_reports) {
+        nr_timer_setup(&l3_measurements->periodical_report_timer,
+                       l3_measurements->periodical_report_interval_ms,
+                       10);
+        nr_timer_start(&l3_measurements->periodical_report_timer);
+      }
+    }
   }
 }
 
