diff --git a/docker/scripts/get-yangs.sh b/docker/scripts/get-yangs.sh
index f391dac..6e321bc 100755
--- a/docker/scripts/get-yangs.sh
+++ b/docker/scripts/get-yangs.sh
@@ -73,3 +73,8 @@ for file in "${special_files[@]}"
 do
    cp "$DIR_AVAILABLE_YANGS/$file" "$DIR_YANGS/"
 done
+
+# Copy vendor-specific YANGs into the yang directory
+if [ -f "$DIRBIN/oai-handover.yang" ]; then
+    cp "$DIRBIN/oai-handover.yang" "$DIR_YANGS/"
+fi
\ No newline at end of file
