diff --git a/openair2/LAYER2/NR_MAC_gNB/gNB_scheduler_dlsch.c b/openair2/LAYER2/NR_MAC_gNB/gNB_scheduler_dlsch.c
index be4850f412..08489925f0 100644
--- a/openair2/LAYER2/NR_MAC_gNB/gNB_scheduler_dlsch.c
+++ b/openair2/LAYER2/NR_MAC_gNB/gNB_scheduler_dlsch.c
@@ -360,6 +360,9 @@ void abort_nr_dl_harq(NR_UE_info_t* UE, int8_t harq_pid)
 
   finish_nr_dl_harq(sched_ctrl, harq_pid);
   UE->mac_stats.dl.errors++;
+
+  NR_UE_harq_t *harq = &sched_ctrl->harq_processes[harq_pid];
+  UE->mac_stats.dl.sdu_errors += harq->num_sdus; // SDU-level failures
 }
 
 bwp_info_t get_pdsch_bwp_start_size(gNB_MAC_INST *nr_mac, NR_UE_info_t *UE)
@@ -1161,6 +1164,7 @@ static void generate_dl_mac_pdu(gNB_MAC_INST *mac,
     mac->mac_stats.dl.used_prb_aggregate += sched_pdsch->rbSize;
 
     /* save retransmission information */
+    harq->num_sdus = sdus;
     harq->sched_pdsch = *sched_pdsch;
     /* save which time allocation has been used, to be used on
      * retransmissions */
