diff --git a/openair2/LAYER2/NR_MAC_gNB/nr_mac_gNB.h b/openair2/LAYER2/NR_MAC_gNB/nr_mac_gNB.h
index e7b3479200..e141efa2e4 100644
--- a/openair2/LAYER2/NR_MAC_gNB/nr_mac_gNB.h
+++ b/openair2/LAYER2/NR_MAC_gNB/nr_mac_gNB.h
@@ -551,6 +551,8 @@ typedef struct NR_UE_harq {
   bool start_tci_timer;
   /// sched_pdsch keeps information on MCS etc used for the initial transmission
   NR_sched_pdsch_t sched_pdsch;
+  // number of SDUs associated with this HARQ process
+  int num_sdus;
 } NR_UE_harq_t;
 
 //! fixme : need to enhace for the multiple TB CQI report
@@ -755,6 +757,7 @@ typedef struct NR_mac_dir_stats {
   uint32_t current_rbs;
   uint64_t prev_sdu_bytes;
   frame_t last_goodput_frame;
+  uint32_t sdu_errors;
 } NR_mac_dir_stats_t;
 
 typedef struct NR_mac_stats {
